### DISCUSSION TITLE
xray的什么功能会随机监听一个UDP端口

### DISCUSSION THREAD

**Discussion by qcxiang** [XTLS/Xray-core#2012] [Q&A]
这两天用命令“ss -tunlp|grep xray”观察到一个现象：
udp    UNCONN 0      0                     *:39745            *:*    users:(("xray",pid=501,fd=3))
udp   UNCONN 0      0                      *:39767            *:*    users:(("xray",pid=501,fd=10))
udp   UNCONN 0      0                      *:38152            *:*    users:(("xray",pid=501,fd=34))
即间隔一会就会启动监听一个随机的UDP端口。

我使用XRAY的方式是通过caddy反代，xray监听的uinx domain socket，按照我肤浅的理解xray是不需要再监听其它TCP或者UDP的端口了。

服务端配置如下：
**00_log.json**
{
  "log": {
    "access": "/var/log/xray-access.log",
    "error": "/var/log/xray.log",
    "loglevel": "warning",
    "dnsLog": true
  }
}

**01_dns.json**
{
  "dns": {
    "hosts": {
      "dns.google": ["8.8.8.8", "8.8.4.4"],
      "cloudflare-dns.com": ["1.1.1.1", "1.0.0.1"],
      "geosite:category-ads-all": "127.0.0.1"
    },
    "servers": [
      {
        "address": "https+local://dns.google/dns-query",
        "domains": ["geosite:geolocation-!cn"],
        "skipFallback": true
      },
      {
        "address": "https+local://cloudflare-dns.com/dns-query",
        "skipFallback": false
      }
    ],
    "queryStrategy": "UseIPv4",
    "disableCache": false,
    "disableFallback": false,
    "disableFallbackIfMatch": true
  }
}

**02_inbounds.json**
{
  "inbounds": [
    {
      "listen": "@xxxxxxxxxx",
      "protocol": "vless",
      "settings": {
        "clients": [
          {
            "id": "xxxxxxxxxxx"
          }
        ],
        "decryption": "none"
      },
      "streamSettings": {
        "network": "h2",
        "security": "none",
        "httpSettings": {
          "host": ["xxx.xxx"],
          "path": "/xxx"
        }
      },
      "sniffing": {
        "enabled": true,
        "destOverride": ["http", "tls"]
      }
    }
  ]
}

**03_routing.json**
{
  "routing": {
    "domainStrategy": "UseIp",
    "domainMatcher": "hybrid",
    "rules": [
      {
        "type": "field",
        "ip": ["geoip:private", "geoip:cn"],
        "outboundTag": "Block"
      },
      {
        "type": "field",
        "outboundTag": "Out-direct",
        "network": "tcp,udp"
      }
    ]
  }
}

**04_outbounds.json**
{
  "outbounds": [
    {
      "tag": "Out-direct",
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIPv4"
      }
    },
    {
      "tag": "Block",
      "protocol": "blackhole",
      "settings": {
        "response": {
          "type": "none"
        }
      }
    }
  ]
}

**Comment by RPRX** [comment]
“代理 UDP 流量”的功能
